File: src/layer_04_infrastructure/ui/web/frontend/hooks/use_create_profile_template.js
import { CreateProfileTemplateController } from "../../../layer_03_interface_adapters/controllers/web/create_profile_template.js";

/**
 * Custom Hook quản lý trạng thái hiển thị và luồng công việc cho tính năng CreateProfileTemplate.
 */
export class UseCreateProfileTemplate extends EventTarget {
    constructor() {
        super();
        this.loading = false;
        this.data = {};
    }
}

File: src/layer_04_infrastructure/ui/desktop_qt6/hooks/use_create_profile_template.py
from PyQt6.QtCore import QObject, pyqtSignal
from src.layer_03_interface_adapters.controllers.desktop.create_profile_template import (
    CreateProfileTemplateController,
)


class UseCreateProfileTemplate(QObject):
    """
    Custom Hook quản lý trạng thái hiển thị và luồng công việc cho tính năng CreateProfileTemplate.
    """

    data_changed = pyqtSignal(dict)
    loading = pyqtSignal(bool)
    error = pyqtSignal(str)

    def __init__(self, parent=None):
        super().__init__(parent)
